One of the CI builds is failing due to:
```
> Task :applyConfiguration FAILED
Unable to initialize logging
thread 'main' panicked at 'Unable to decrypt and copy files: SecretsNotPresent', src/configure.rs:193:53
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':applyConfiguration'.
> Process 'command '/home/circleci/project/vendor/configure/configure'' finished with non-zero exit value 101
```
This seems something not specific to my changes, but I can be wrong. If there is something I need to do to fix it, please let me know.

--- a/example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_ThemeTestJetpack.java
+++ b/example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_ThemeTestJetpack.java
@@ -128,7 +128,16 @@ public class ReleaseStack_ThemeTestJetpack extends ReleaseStack_Base {
         assertNotNull(themeToActivate);
 
         // activate it
-        ThemeModel activatedTheme = activateTheme(jetpackSite, themeToActivate);
+        ThemeModel activatedTheme = activateTheme(jetpackSite, themeToActivate, false);
+        assertNotNull(activatedTheme);
+        assertEquals(activatedTheme.getThemeId(), themeToActivate.getThemeId());
+
+        // Select again
+        themeToActivate = getOtherTheme(themes, currentTheme.getThemeId());
+        assertNotNull(themeToActivate);
+
+        // activate it
+        activatedTheme = activateTheme(jetpackSite, themeToActivate, true);
         assertNotNull(activatedTheme);
         assertEquals(activatedTheme.getThemeId(), themeToActivate.getThemeId());
 
@@ -401,7 +410,14 @@ public class ReleaseStack_ThemeTestJetpack extends ReleaseStack_Base {
         return mThemeStore.getActiveThemeForSite(jetpackSite);
     }
 
-    private ThemeModel activateTheme(@NonNull SiteModel jetpackSite, @NonNull ThemeModel themeToActivate)
+    private ThemeModel activateTheme(@NonNull SiteModel jetpackSite,
+                                     @NonNull ThemeModel themeToActivate) throws InterruptedException {
+        return activateTheme(jetpackSite, themeToActivate, true);
+    }
+
+    private ThemeModel activateTheme(@NonNull SiteModel jetpackSite,
+                                     @NonNull ThemeModel themeToActivate,
+                                     boolean dontChangeHomepage)
             throws InterruptedException {
         mCountDownLatch = new CountDownLatch(1);
         mNextEvent = TestEvents.ACTIVATED_THEME;

--- a/example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_ThemeTestJetpack.java
+++ b/example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_ThemeTestJetpack.java
@@ -109,7 +109,34 @@ public class ReleaseStack_ThemeTestJetpack extends ReleaseStack_Base {
     }
 
     @Test
-    public void testActivateTheme() throws InterruptedException {
+    public void testActivateThemeByKeepingHomepage() throws InterruptedException {
+        final SiteModel jetpackSite = signIntoWpComAccountWithJetpackSite();
+
+        // fetch installed themes
+        fetchInstalledThemes(jetpackSite);
+
+        // make sure there are at least 2 themes, one that's active and one that will be activated
+        List<ThemeModel> themes = mThemeStore.getThemesForSite(jetpackSite);
+        assertTrue(themes.size() > 1);
+
+        // fetch active theme
+        ThemeModel currentTheme = fetchCurrentTheme(jetpackSite);
+        assertNotNull(currentTheme);
+
+        // select a different theme to activate
+        ThemeModel themeToActivate = getOtherTheme(themes, currentTheme.getThemeId());
+        assertNotNull(themeToActivate);
+
+        // activate it
+        ThemeModel activatedTheme = activateTheme(jetpackSite, themeToActivate, true);
+        assertNotNull(activatedTheme);
+        assertEquals(activatedTheme.getThemeId(), themeToActivate.getThemeId());
+
+        signOutWPCom();
+    }
+
+    @Test
+    public void testActivateThemeByUsingThemeHomepage() throws InterruptedException {
         final SiteModel jetpackSite = signIntoWpComAccountWithJetpackSite();
 
         // fetch installed themes
@@ -129,15 +156,6 @@ public class ReleaseStack_ThemeTestJetpack extends ReleaseStack_Base {
 
         // activate it
         ThemeModel activatedTheme = activateTheme(jetpackSite, themeToActivate, false);
-        assertNotNull(activatedTheme);
-        assertEquals(activatedTheme.getThemeId(), themeToActivate.getThemeId());
-
-        // Select again
-        themeToActivate = getOtherTheme(themes, currentTheme.getThemeId());
-        assertNotNull(themeToActivate);
-
-        // activate it
-        activatedTheme = activateTheme(jetpackSite, themeToActivate, true);
         assertNotNull(activatedTheme);
         assertEquals(activatedTheme.getThemeId(), themeToActivate.getThemeId());
 
@@ -421,7 +439,7 @@ public class ReleaseStack_ThemeTestJetpack extends ReleaseStack_Base {
             throws InterruptedException {
         mCountDownLatch = new CountDownLatch(1);
         mNextEvent = TestEvents.ACTIVATED_THEME;
-        ActivateThemePayload payload = new ActivateThemePayload(jetpackSite, themeToActivate);
+        ActivateThemePayload payload = new ActivateThemePayload(jetpackSite, themeToActivate, dontChangeHomepage);
         mDispatcher.dispatch(ThemeActionBuilder.newActivateThemeAction(payload));
         assertTrue(mCountDownLatch.await(TestUtils.DEFAULT_TIMEOUT_MS, TimeUnit.MILLISECONDS));
 

--- a/example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_ThemeTestWPCom.java
+++ b/example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_ThemeTestWPCom.java
@@ -110,7 +110,7 @@ public class ReleaseStack_ThemeTestWPCom extends ReleaseStack_WPComBase {
         ThemeModel themeToActivate = getTwentySomethingFreeTheme(currentTheme.getThemeId(),
                 mThemeStore.getWpComThemes());
         assertNotNull(themeToActivate);
-        activateTheme(themeToActivate, true);
+        activateTheme(themeToActivate, false);
 
         // Assert that the activation was successful
         ThemeModel activatedTheme = mThemeStore.getActiveThemeForSite(sSite);

--- a/example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_ThemeTestWPCom.java
+++ b/example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_ThemeTestWPCom.java
@@ -75,7 +75,7 @@ public class ReleaseStack_ThemeTestWPCom extends ReleaseStack_WPComBase {
     }
 
     @Test
-    public void testActivateTheme() throws InterruptedException {
+    public void testActivateThemeByKeepingHomepage() throws InterruptedException {
         // Make sure no theme is active at first
         assertNull(mThemeStore.getActiveThemeForSite(sSite));
         ThemeModel currentTheme = fetchCurrentTheme();
@@ -88,7 +88,29 @@ public class ReleaseStack_ThemeTestWPCom extends ReleaseStack_WPComBase {
         ThemeModel themeToActivate = getTwentySomethingFreeTheme(currentTheme.getThemeId(),
                 mThemeStore.getWpComThemes());
         assertNotNull(themeToActivate);
-        activateTheme(themeToActivate);
+        activateTheme(themeToActivate, true);
+
+        // Assert that the activation was successful
+        ThemeModel activatedTheme = mThemeStore.getActiveThemeForSite(sSite);
+        assertNotNull(activatedTheme);
+        assertEquals(activatedTheme.getThemeId(), themeToActivate.getThemeId());
+    }
+
+    @Test
+    public void testActivateThemeByUsingThemeHomepage() throws InterruptedException {
+        // Make sure no theme is active at first
+        assertNull(mThemeStore.getActiveThemeForSite(sSite));
+        ThemeModel currentTheme = fetchCurrentTheme();
+        assertNotNull(currentTheme);
+
+        // Fetch wp.com themes to activate a different theme
+        fetchWpComThemes();
+
+        // activate a different "Twenty ..." theme
+        ThemeModel themeToActivate = getTwentySomethingFreeTheme(currentTheme.getThemeId(),
+                mThemeStore.getWpComThemes());
+        assertNotNull(themeToActivate);
+        activateTheme(themeToActivate, true);
 
         // Assert that the activation was successful
         ThemeModel activatedTheme = mThemeStore.getActiveThemeForSite(sSite);
@@ -210,10 +232,11 @@ public class ReleaseStack_ThemeTestWPCom extends ReleaseStack_WPComBase {
         assertTrue(mCountDownLatch.await(TestUtils.DEFAULT_TIMEOUT_MS, TimeUnit.MILLISECONDS));
     }
 
-    private void activateTheme(ThemeModel themeToActivate) throws InterruptedException {
+    private void activateTheme(ThemeModel themeToActivate,
+                               boolean dontChangeHomePage) throws InterruptedException {
         mCountDownLatch = new CountDownLatch(1);
         mNextEvent = TestEvents.ACTIVATED_THEME;
-        ActivateThemePayload payload = new ActivateThemePayload(sSite, themeToActivate);
+        ActivateThemePayload payload = new ActivateThemePayload(sSite, themeToActivate, dontChangeHomePage);
         mDispatcher.dispatch(ThemeActionBuilder.newActivateThemeAction(payload));
         assertTrue(mCountDownLatch.await(TestUtils.DEFAULT_TIMEOUT_MS, TimeUnit.MILLISECONDS));
     }

